test/positivity.lean

```diff
@@ -103,3 +104,18 @@ example {a : ℤ} (ha : a > 0) : 0 ≤ a := by positivity
 example {a : ℤ} (ha : 0 < a) : a ≥ 0 := by positivity
 
 example {a : ℤ} (ha : a > 0) : a ≥ 0 := by positivity
+
+/- ## Coercions -/
+
+open_locale nnreal
+
+example {n : ℕ} : 0 ≤ n := by positivity
+example {n : ℕ} : 0 ≤ (n : ℤ) := by positivity
+example {n : ℕ} : 0 ≤ (n : ℚ) := by positivity
+example {n : ℕ} : 0 ≤ ((n : ℚ) : ℝ) := by positivity
```

Comment:
Those two fail and I don't understand why.

Reply:
Could you show the error with `set_option pp.all true`?

Reply:
Ahah, found it. Another day, another time where it would have been useful to be able to inspect the error state.

Somehow, it came up with a proof for `nnreal` rather than `rat`.

Reply:
I fixed it by replacing `mk_app` with `mk_mapp` for this one case. Would it be better to do so everywhere or is this case somehow special because I'm starting a proof of nonnegativity (rather than passing one around)?

Reply:
I'm not aware of a reason that `mk_mapp` works better here so as far as I know you don't need to switch. Probably it has better metavariable handling? Maybe you should also add a test case for cast from `ℝ` to something that is not `ereal`, to be sure that cast works correctly too. (It's fine if you test that the tactic fails because it can't find a proof, as long as it's not a type error in the kernel.)

Reply:
Is there any cast from `ℝ` to something that isn't `ereal`?

Reply:
Maybe there isn't anything ordered except for `ℂ`, indeed. It's fine if the positivity lemmas are missing, it should be possible to run the tactic, succeed if that fails, and then type check the resulting proof, failing if the type check fails.

Reply:
I am failing to fail the failure of understanding your sentence. 😁

Reply:
I guess the conclusion is: this isn't happening in practice so why care?